packages/shared/src/stringInterpolation.ts

```diff
@@ -36,8 +36,10 @@ export const createStringInterpolator = (
 ) => {
   if (!namedColorRegex)
     namedColorRegex = G.colors
-      ? new RegExp(`(${Object.keys(G.colors).join('|')})`, 'g')
-      : /^\b$/ // never match
+      ? // match color names, ignore partial matches
+        new RegExp(`(?<!\\w)(${Object.keys(G.colors).join('|')})(?!\\w)`, 'g')
```

Comment:
To clarify, the first group in the regex `(?<!\\w)` is what I mention in regressions.